fix: add functional tests for rails script + helpers

benkoshy opened the pull request:
### Why do we need this?

Because, as a rails user, when debugging #341 (https://github.com/ddnexus/pagy/issues/341)
I wanted to see a functional test - because it is familiar to rails users, and makes it super easy to debug.

Also small additional changes have been made to facilitate this - byebug as a debugging tool, and I have copied the script used by rails and made minor modifications where appropriate.

I hope it is useful

ddnexus commented:
@benkoshy I enabled the rubocop Style/Documentation cop, so it will complain if we miss some class/module documentation comments. You can also add a `# :nodoc:` on the same definition line if you want to suppress it.

benkoshy commented on apps/rails_inline_output.rb:
```diff
@@ -13,40 +14,81 @@
+# require 'pagy/extras/support'
+# require 'pagy/extras/trim'
+# require 'pagy/extras/uikit'
+
```
I'll add those examples directly in the code, so people have a foundation and can modify/remove as required

benkoshy commented:
Chrs for the review. 

Changes made and force pushed:

* Made the changes requested in #342 discussion - with slight modifications. 
* Added links - to make it easy for non-minitest users to set up assertions.
* Simplified instructions to be more concise.

I hope this helps, and the community finds it useful.

ddnexus commented:
I couldn't get it cleanly merged by GitHib, so I did it manually.
